### PR TITLE
chore(actions): enforce new changelog format and convention

### PR DESCRIPTION
**Pull request by dndx**
This ticket make sure new changelog entries submitted adheres to the [changelog standard](https://github.com/Kong/gateway-changelog) to avoid conflicts like `CHANGELOG` from slipping in.

### CODE REVIEW

**Comment by dndx** [comment]
I checked the results with some fake commits (https://github.com/Kong/kong/pull/11757/commits), these checks seems to work.

File: .github/workflows/changelog-requirement.yml [hunk 1]
```diff
@@ -18,15 +19,49 @@ jobs:
...
         run: >
-          if [ "${{ steps.changelog-check.outputs.all_changed_and_modified_files_count }}" = "0" ]; then
-            echo "Should contain at least one changelog file in changelog/unreleased/*/ directory"
-            exit 1
-          fi
+          echo "Changelog file expected but found none. If you believe this PR requires no changelog entry, label it with \"skip-changelog\"."
+          echo "Refer to https://github.com/Kong/gateway-changelog for format guidelines."
+          exit 1
+
+      - name: Check incorrect case for changelog directory
```
Comment (hanshuebner) [inline .github/workflows/changelog-requirement.yml:41]:
```suggestion
      - name: Check correct case for changelog directory
```

File: .github/workflows/changelog-requirement.yml [hunk 2]
```diff
@@ -18,15 +19,49 @@ jobs:
...
+        if: steps.changelog-list.outputs.upper_case_any_changed == 'true'
+        run: |
+          echo "Please use \"changelog\" (all lowercase) for changelog modifications."
+          echo "Refer to https://github.com/Kong/gateway-changelog for format guidelines."
+          echo "Bad file(s): ${{ steps.changelog-list.outputs.upper_case_all_changed_files }}"
+          exit 1
+
+      - name: Check numbered filename for changelog entry
```
Comment (hanshuebner) [inline .github/workflows/changelog-requirement.yml:49]:
```suggestion
      - name: Check descriptive filename for changelog entry
```

File: .github/workflows/changelog-requirement.yml [hunk 3]
```diff
@@ -18,15 +19,49 @@ jobs:
...
+        if: steps.changelog-list.outputs.numbered_any_changed == 'true'
+        run: |
+          echo "Please use short descriptive name for changelog files instead of numbers."
+          echo "E.g. bump_openresty.yml instead of 12345.yml."
+          echo "Refer to https://github.com/Kong/gateway-changelog for format guidelines."
+          echo "Bad file(s): ${{ steps.changelog-list.outputs.numbered_all_changed_files }}"
+          exit 1
+
+      - name: Check deprecated YAML keys
```
Comment (hanshuebner) [inline .github/workflows/changelog-requirement.yml:58]:
```suggestion
      - name: Fail when deprecated YAML keys are used
```

**Review by hanshuebner — APPROVED** [review]

**Pull request merged**